### PR TITLE
Embed related item IDs into view results

### PR DESCRIPTION
**Pull request by dwhitena**
# What this PR does:
- embeds related item IDs, by tag, into view items under a field called `related`.
- automatically extracts the related IDs from the graph path and tags.
- uses an alias for each path in a view to make sure IDs are embedded in relevant items.

# How to test this PR:
- `go test ./...`

### CODE REVIEW

File: internal/wire/wire.go [hunk 1]
```diff
@@ -315,11 +364,84 @@ func viewIDs(v *view.View, path *path.Path, key string, graphDB *cayley.Handle)
...
+
+		// If the order is greater than 1, we should embed at the next tagged
+		// level up from the item.
+		var embedTag string
+		var ok bool
+		if order > 1 {
+
+			// Get the tag of the item in which the embed should be placed.
+			for order > 1 {
+				alias := aliasOrder[0] + "_" + strconv.Itoa(order-1)
```
Comment (wyattjoh) [inline internal/wire/wire.go]: Should still check that aliasOrder has a length greater than 0, `strings.Split` can still return an empty slice.

File: internal/wire/wire_test.go [hunk 2]
```diff
@@ -269,10 +269,28 @@ func TestExecuteSplitPath(t *testing.T) {
...
+				if !ok {
+					continue
+				}
+				itemID, ok := itemField.(string)
+				if !ok {
+					continue
+				}
+				if itemID == wirePrefix+"a63af637-58af-472b-98c7-f5c00743bac6" {
+					_, ok := itm["related"]
+					if !ok {
```
Comment (wyattjoh) [inline internal/wire/wire_test.go]: I'd move this to an inline if check:

```go
if _, ok := itm["related"]; !ok {
  ...
}
```

**Comment by gabelula** [comment]
Update localhost.cfg to have  WIRE_MONGO_URI instead of WIRE_MONGO_HOST.

File: internal/wire/wire.go [hunk 3]
```diff
@@ -82,15 +84,15 @@ func Execute(context interface{}, mgoDB *db.DB, graphDB *cayley.Handle, viewPara
 	}
 
 	// Retrieve the item IDs for the view.
-	ids, err := viewIDs(v, graphPath, viewParams.ItemKey, graphDB)
+	ids, embeds, err := viewIDs(v, graphPath, viewParams.ItemKey, graphDB)
```
Comment (gabelula) [inline internal/wire/wire.go]: Can we change the comment here to explain what embeds are?

Comment (dwhitena) [inline internal/wire/wire.go]: Fixed here: https://github.com/coralproject/shelf/pull/287/commits/cb9737de9dc3ca27905710d774038e96193235df

**Comment by dwhitena** [comment]
@gabelula the localhost.cfg is fixed here: https://github.com/coralproject/shelf/pull/287/commits/cb9737de9dc3ca27905710d774038e96193235df

**Comment by dwhitena** [comment]
@gabelula After merging in from master, I realized we have done away with the localhost.cfg.  I changed back to XENIA as the prefix in the wire tests, as wire will never run on it's own (and thus will use the XENIA env vars anyway).

**Comment by jde** [comment]
Looks good to me. Merging.

**Pull request merged**